app/services/discourse_translator/discourse_ai.rb

```diff
@@ -33,7 +30,12 @@ def self.translate!(translatable, target_locale_sym = I18n.locale)
                 ),
               )
       end
-      ::DiscourseAi::Translator.new(text_for_translation(translatable), target_locale_sym).translate
+
+      text = text_for_translation(translatable)
+      chunks = DiscourseTranslator::ContentSplitter.split(text)
+      chunks
+        .map { |chunk| ::DiscourseAi::Translator.new(chunk, target_locale_sym).translate }
```

Comment:
Not sure if you considered it but I'm wondering if we should translate chunks in parallel.